routemaster/mixins/redis.rb

```diff
@@ -2,13 +2,31 @@
 require 'redis'
 require 'redis/connection/hiredis'
 
-module Routemaster::Mixins::Redis
-  def self.included(by)
-    by.extend(self)
-    by.send(:protected, :_redis)
-  end
+module Routemaster
+  module Mixins
+    module Redis
+      def self.included(by)
+        by.extend(self)
+        by.send(:protected, :_redis)
+      end
+
+      def _redis
+        @@_redis ||= ::Redis.new(url: ENV['ROUTEMASTER_REDIS_URL'])
+      end
+
+      def _lua_sha(script_name)
+        @@_lua_sha ||= {}
+        sha = @@_lua_sha[script_name] and return sha
+
+        path = File.expand_path("../../lua/#{script_name}.lua", __FILE__)
+        sha = _redis.script('load', File.read(path))
+        @@_lua_sha[script_name] = sha
+      end
 
-  def _redis
-    @@_redis ||= Redis.new(url: ENV['ROUTEMASTER_REDIS_URL'])
+      def _lua_run(script_name, keys:nil, argv:nil)
```

Comment:
Would it make sense to name this something like `_redis_lua_run` as once mixed in the name `_lua_run` won't really give much context about _where_ the Lua is being run?